lib/Sema/SemaType.cpp

```diff
@@ -3717,6 +3717,22 @@ QualType Sema::MakeCheckedArrayType(QualType T, bool Diagnose,
 /// If a declaration has a Checked C bounds-safe interface attached to it,
 /// construct and return the checked type for the interface. Otherwise
 /// return a null QualType.
+QualType Sema::GetCheckedCInteropType(const DeclaratorDecl *Decl) {
+  const BoundsExpr *Bounds = Decl->getBoundsExpr();
+  if (!Bounds)
+    return QualType();
+
+  QualType Ty;
+  if (const FunctionDecl *FD = dyn_cast<FunctionDecl>(Decl))
+    Ty = FD->getReturnType();
```

Comment:
Why do we return only the return type, don't we want the whole function type?

Reply:
Good catch - I agree this is confusing.  The prior code wanted the return type, and this was left over from that use.   I looked more closely and saw that there was only one other method calling this method.  It wasn't calling this on FunctionDecls anyway.   I cleaned up the code by deleting this method and folding its functionality its caller.